includes/class-llms-order-generator.php

```diff
@@ -0,0 +1,732 @@
+				return $this->error( self::E_COUPON_INVALID, $valid->get_error_message() );
+			}
+
+			$this->coupon = $coupon;
+
+		}
+
+		return true;
+	}
+
+	/**
+	 * Validates the payment gateway.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean|WP_Error Returns `true` on success or an error object.
+	 */
+	protected function validate_gateway() {
+
+		$coupon_id = $this->coupon ? $this->coupon->get( 'id' ) : null;
+
+		/**
+		 * If payment is required, verify we have a gateway.
+		 *
+		 * For free plans the manual gateway is automatically used, whether or not it's enabled.
+		 */
+		if ( $this->plan->requires_payment( $coupon_id ) && empty( $this->data['llms_payment_gateway'] ) ) {
+			return $this->error( self::E_GATEWAY_REQUIRED, __( 'No payment method selected.', 'lifterlms' ) );
+		}
+
+		$gateway_id = $this->data['llms_payment_gateway'] ?? 'manual';
+		$is_valid   = llms_can_gateway_be_used_for_plan( $gateway_id, $this->plan );
+		if ( is_wp_error( $is_valid ) ) {
+			return $is_valid;
+		}
+
+		$this->gateway = llms()->payment_gateways()->get_gateway_by_id( $gateway_id );
+		return true;
+
+	}
+
+	/**
+	 * Validates the order.
+	 *
+	 * Ensures the submitted order key is valid and that the order can be confirmed.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean|WP_Error Returns `true` on success or an error object.
+	 */
+	protected function validate_order() {
+
+		$order_id = $this->get_order_id();
+
+		if ( 'new' === $order_id || 'llms_order' !== get_post_type( $order_id ) ) {
+			return $this->error(
+				self::E_ORDER_NOT_FOUND,
+				__( 'Could not locate an order to confirm.', 'lifterlms' )
+			);
+		}
+
+		$order = llms_get_post( $order_id );
+		if ( ! $order->can_be_confirmed() ) {
+			return $this->error(
+				self::E_ORDER_NOT_CONFIRMABLE,
+				__( 'Could not locate an order to confirm.', 'lifterlms' )
+			);
+		}
+
+		$this->order = $order;
+		return true;
+
+	}
+
+	/**
+	 * Validates the access plan.
+	 *
+	 * Ensures the access plan data was submitted and that it's a valid plan.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean|WP_Error Returns `true` on success or an error object.
+	 */
+	protected function validate_plan() {
+
+		$plan_id = $this->data['llms_plan_id'] ?? null;
+		if ( ! $plan_id ) {
+			return $this->error( self::E_PLAN_REQUIRED, __( 'Missing access plan ID.', 'lifterlms' ) );
+		}
+
+		$plan = llms_get_post( $plan_id );
+		if ( ! $plan || 'llms_access_plan' !== $plan->get( 'type' ) ) {
+			return $this->error( self::E_PLAN_NOT_FOUND, __( 'Access plan not found.', 'lifterlms' ) );
+		}
+
+		$this->plan = $plan;
+		return true;
+
+	}
+
+	/**
+	 * Validates the site's terms and conditions were submitted.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean|WP_Error Returns `true` on success or an error object.
+	 */
+	protected function validate_terms() {
+
+		if ( llms_are_terms_and_conditions_required() && ! llms_parse_bool( $this->data['llms_agree_to_terms'] ?? 'no' ) ) {
```

Comment:
Instead of something like this:
`$this->data['llms_agree_to_terms'] ?? 'no'`
maybe we can use `wp_parse_args()` in the class constructor so to merge passed data with defaults?

Reply:
Yea but why?

Reply:
to reduce the checks

Reply:
even if the default value is set in the constructor we still need to run this check though.

Reducing the need for null coalesce by running `wp_parse_args()` seems unnecessary to me... It's probably more performant this way, don't you think?

Reply:
Yeah, it's more performant, but having defaults set in one place makes me feel better. nevermind. :)